nmdc/messages.go

```diff
@@ -966,6 +967,143 @@ func (m *Error) UnmarshalNMDC(text []byte) error {
 	return nil
 }
 
+type DataType byte
+
+const (
+	DataTypeAnyFileType     = DataType('1')
+	DataTypeAudioFiles      = DataType('2')
+	DataTypeCompressedFiles = DataType('3')
+	DataTypeDocumentFiles   = DataType('4')
+	DataTypeExecutableFiles = DataType('5')
+	DataTypePictureFiles    = DataType('6')
+	DataTypeVideoFiles      = DataType('7')
+	DataTypeFolders         = DataType('8')
+	DataTypeTTH             = DataType('9')
+)
+
+type Search struct {
+	Address        string
+	Nick           Name
+	SizeRestricted bool
+	IsMaxSize      bool
+	Size           uint64
+	DataType       DataType
+	SearchPattern  string
```

Comment:
```suggestion
	Pattern  string
```
It's already in `Search` struct, so you can omit this prefix.